### PR TITLE
have the gui stop the service via RPC instead of shelling out

### PR DESCRIPTION
**Pull request by zanderz**
I can't test this yet because the widget menu hamburger has disappeared on me.
UPDATE: tested it, it works. Hamburger problem is a separate GUI issue.

### CODE REVIEW

**Comment by maxtaco** [comment]
Can’t be me but why this change? Is there a jira ticket?

**Comment by zanderz** [comment]
No, I just wanted to make this right for Surya.

**Comment by heronhaye** [comment]
So this will call the ctl.Stop RPC? Ok, looks good to me (idk if you want someone on frontend to review as well though). I'll fixup my PR after this is merged

**Comment by zanderz** [comment]
This works for me now but it should have an official desktop review.

File: shared/actions/settings.js [hunk 1]
```diff
@@ -479,6 +479,9 @@ const loadHasRandomPW = state =>
 // Mark that we are not randomPW anymore if we got a password change.
 const passwordChanged = () => SettingsGen.createLoadedHasRandomPw({randomPW: false})
 
+// $FlowIssue
+const stop = (_, action) => RPCTypes.ctlStopRpcPromise(action.payload.exitCode)
```
Comment (buoyad) [inline shared/actions/settings.js]: I think flow complains because the param should be `{exitCode: action.payload.exitCode}`

Comment (zanderz) [reply shared/actions/settings.js]: whoops, thanks again Danny!
This should wait till after code freeze though right?

Comment (heronhaye) [reply shared/actions/settings.js]: I was kind of hoping to get this in for the release. But if it's risky we can bump it to the next one.

Comment (buoyad) [reply shared/actions/settings.js]: Seems a little risky, I'd say let it wait

**Pull request merged**